components/authentication-framework/org.wso2.carbon.identity.application.authentication.endpoint/src/main/webapp/basicauth.jsp

```diff
@@ -160,42 +164,40 @@
                 identityMgtEndpointContext = IdentityUtil.getServerURL("/accountrecoveryendpoint", true, true);
             }
 
-            URL url = null;
-            HttpURLConnection httpURLConnection = null;
+            HttpClient client = new HttpClient();
+            if ("https".equals(scheme)) {
```

Comment:
No. The one packed by the kernel initiates a Protocol with SSLProtocolSocketFactory only if there are no other protocols registered for https (Check [1]). In this case another Protocol is already registered with org.opensaml.ws.soap.client.http.TLSProtocolSocketFactory. This is done by opensaml library. TLSProtocolSocketFactory has a Strict hostname verification. We can actually turn that off with **-Dorg.opensaml.httpclient.https.disableHostnameVerification=true** property. But I think it's better to handle the hostname verification with one property. This code section makes sure that SSLProtocolSocketFactory is used and hence the httpclient honours **httpclient.hostnameVerifier** property.
[1] https://github.com/wso2/wso2-commons-httpclient/blob/master/commons-httpclient/src/main/java/org/apache/commons/httpclient/protocol/Protocol.java#L114